### PR TITLE
Fix crash when opening email with empty From header

### LINKED ISSUE #1803: Empty From header throws error and Conversations cannot open the message
* **Thunderbird version:** 91.5.0
* **Thunderbird Conversation version:** 4.0.12
* **Operating System:** Ubuntu 20.04

**What did you do?**

Receive a message with an empty `From` header and try to open it.

**What did you expect to happen?**

The message to be displayed.

**What actually happened?**

A blank page is shown:

image

In the console:
```
Uncaught (in promise) TypeError: from is undefined
    MessageHeader chrome://conversations/content/239.bundle.js:3183
    Ch chrome://conversations/content/465.bundle.js:373
    ck chrome://conversations/content/465.bundle.js:483
    bk chrome://conversations/content/465.bundle.js:466
    ak chrome://conversations/content/465.bundle.js:466
    Tj chrome://conversations/content/465.bundle.js:466
    Lj chrome://conversations/content/465.bundle.js:459
    jg chrome://conversations/content/465.bundle.js:339
    unstable_runWithPriority chrome://conversations/content/465.bundle.js:570
    gg chrome://conversations/content/465.bundle.js:338
    jg chrome://conversations/content/465.bundle.js:339
    ig chrome://conversations/content/465.bundle.js:338
    Wj chrome://conversations/content/465.bundle.js:460
    notify chrome://conversations/content/367.bundle.js:366
    notifyNestedSubs chrome://conversations/content/367.bundle.js:436
    handleChangeWrapper chrome://conversations/content/367.bundle.js:441
    dispatch chrome://conversations/content/367.bundle.js:970
    middleware chrome://conversations/content/367.bundle.js:590
    dispatch chrome://conversations/content/367.bundle.js:1320
    updateConversation chrome://conversations/content/239.bundle.js:1426
```

Which is this line:
```javascript
  })), /*#__PURE__*/react.createElement(Avatar, {
    url: from.avatar,
    style: from.colorStyle,
    initials: from.initials
  }),
```
https://github.com/thunderbird-conversations/thunderbird-conversations/blob/v4.0.12/addon/content/components/message/messageHeader.jsx#L339

Reproducible with `sendmail`:
```
sendmail -f <smtp.from> <smtp.to> <<EOF
> From:
> To: <header.to>
> Subject: Test
>
> Test
> EOF
```

### PR DESCRIPTION
**Pull request by simon816**
Fixes #1803 

image

I had a poke around with the email and found it would crash with the "Forward this conversation" button so fixed that too.

### CODE REVIEW

File: addon/content/reducer/conversationUtils.js [hunk 1]
```diff
@@ -65,26 +65,33 @@ export let conversationUtils = new (class {
...
+              '<b><span><a style="color: ',
+              msg.from.colorStyle.backgroundColor,
+              ' !important; text-decoration: none !important; font-weight: bold" href="mailto:',
+              msg.from.email,
+              '">',
+              this._escapeHtml(msg.from.name),
+            ]
+          : [],
+        [
+          "</a></span></b><br />",
```
Comment (Standard8) [inline addon/content/reducer/conversationUtils.js:84]: Please can you move this line to within the array of the `msg.from` check?

We don't need to close those tags in the case of the missing from, and I think the `br` is unnecessary as well.

**Review by Standard8 — CHANGES_REQUESTED** [review]
Thank you for the patch, I'm sorry this has taken a while for me to get to it. Please can you update it with the change I suggested inline.

File: addon/content/reducer/conversationUtils.js (hunk 1 above)
Comment (simon816) [reply addon/content/reducer/conversationUtils.js]: Oh yeah, somehow missed this. Have fixed

**Review by Standard8 — APPROVED** [review]
Thank you for the update, this looks good - thank you for working on this.

**Pull request merged**